app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -82,36 +92,48 @@ class EISConnectorImpl(
         Future.failed(new IllegalStateException(s"An unexpected error occurred - got a ${x.getClass}"))
     }
 
-  override def post(body: Source[ByteString, _], hc: HeaderCarrier): Future[Either[RoutingError, Unit]] =
+  private def extractHeaders(movementId: MovementId, messageId: MessageId, hc: HeaderCarrier): Seq[(String, String)] = {
+    val dateHeader: Seq[(String, String)] = hc.headers(Seq(HeaderNames.DATE)) match {
+      case Seq() => Seq(HeaderNames.DATE -> HTTP_DATE_FORMATTER.format(OffsetDateTime.now()))
+      case x     => x
+    }
+
+    val requestHeaders: Seq[(String, String)] = hc.headers(Seq("X-Message-Type")) ++ dateHeader ++ Seq(
+      "X-Correlation-Id"        -> UUID.randomUUID().toString,
+      "CustomProcessHost"       -> "Digital",
+      HeaderNames.CONTENT_TYPE  -> MimeTypes.XML,
+      HeaderNames.ACCEPT        -> MimeTypes.XML,
+      HeaderNames.AUTHORIZATION -> s"Bearer ${eisInstanceConfig.headers.bearerToken}",
+      "X-Conversation-Id"       -> ConversationId(movementId, messageId).value.toString
+    )
+
+    hc.headersForUrl(headerCarrierConfig)(eisInstanceConfig.url) ++ requestHeaders
+
+  }
+
+  override def post(movementId: MovementId, messageId: MessageId, body: Source[ByteString, _], hc: HeaderCarrier): Future[Either[RoutingError, Unit]] =
     retryingOnFailures(
       retries.createRetryPolicy(eisInstanceConfig.retryConfig),
       (t: Either[RoutingError, Unit]) => Future.successful(t.isRight),
       onFailure
     ) {
-      val requestHeaders = hc.headers(OutgoingHeaders.headers) ++ Seq(
-        "X-Correlation-Id"        -> UUID.randomUUID().toString,
-        "CustomProcessHost"       -> "Digital",
-        HeaderNames.ACCEPT        -> MimeTypes.XML, // TODO: is this still relevant? Can't use ContentTypes.XML because EIS will not accept "application/xml; charset=utf-8"
-        HeaderNames.AUTHORIZATION -> s"Bearer ${eisInstanceConfig.headers.bearerToken}"
-      )
-
       implicit val headerCarrier: HeaderCarrier = hc
         .copy(authorization = None, otherHeaders = Seq.empty)
```

Comment:
is this authorization = None is required? I can see we have set that with extractHeaders. So should be removed?

Reply:
Yes, I was seeing duplication of headers when I left it in.